src/bin/tools/unzip.ts

```diff
@@ -1,184 +1,89 @@
-    let chunk = await readFileChunk(zipFile, size - chunkSize, size);
-    const unread = size - chunk.length;
-    let i = chunk.length - 4;
-    let found = false;
-    // Find central directory end header, reading backwards from the end
-    while (!found && i-- > 0) if (chunk[i] === 0x50 && chunk.readUInt32LE(i) === 0x06054b50) found = true;
-    if (!found) throw new Error("Not a zip file");
-    // This method will fail on a multi-disk zip, so bail early.
-    if (chunk.readUInt16LE(i + 4) !== 0) throw new Error("Multi-disk zip not supported");
-    let nFiles = chunk.readUint16LE(i + 10);
-    // Get the position of the central directory
-    const directorySize = chunk.readUint32LE(i + 12);
-    const directoryOffset = chunk.readUint32LE(i + 16);
-    if (directoryOffset === 0xffff_ffff) throw new Error("zip64 not supported");
-    if (directoryOffset > size) throw new Error(`Central directory offset ${directoryOffset} is outside file`);
-    i = directoryOffset - unread;
-    // If i < 0, it means that the central directory is not contained within `chunk`
-    if (i < 0) {
-        chunk = await readFileChunk(zipFile, directoryOffset, directoryOffset + directorySize);
-        i = 0;
-    }
-    // Now iterate the central directory records, yield an `ZipRecord` for every entry
-    while (nFiles-- > 0) {
-        // Check for marker bytes
-        if (chunk.readUInt32LE(i) !== 0x02014b50) throw new Error("No central directory record at position " + (unread + i));
-        const compressionMethod = ({ 8: "deflate" } as const)[chunk.readUint16LE(i + 10)];
-        const compressedFileSize = chunk.readUint32LE(i + 20);
-        const filenameLength = chunk.readUint16LE(i + 28);
-        const extraLength = chunk.readUint16LE(i + 30);
-        const commentLength = chunk.readUint16LE(i + 32);
-        // Start of the actual content byte stream is after the 'local' record header,
-        // which is  30 bytes long plus filename and extra field
-        const start = chunk.readUint32LE(i + 42) + 30 + filenameLength + extraLength;
-        const end = start + compressedFileSize;
-        const filename = chunk.slice(i + 46, i + 46 + filenameLength).toString("utf-8");
-        const createRecordReadStream = () => {
-            const input = createReadStream(zipFile, { start, end });
-            if (compressionMethod === "deflate") {
-                const inflate = createInflateRaw();
-                input.pipe(inflate);
-                return inflate;
-            }
-            return input;
-        };
-        if (end > start) yield { path: filename, createReadStream: createRecordReadStream, compressionMethod };
-        // advance pointer to next central directory entry
-        i += 46 + filenameLength + extraLength + commentLength;
-    }
+                    // Remove the unzipSubPath from the file name
+                    entry.fileName = entry.fileName.substring(unzipSubPath.length);
```

Comment:
not a fan of this mutation

maybe better declare `let target = path.join(targetFolder,entry.filename)` above the if body and correct `target = path.join(targetFolder, entry.filename.substring(unzipSubPath.length))` inside the if body?

or 

put it in an iif? (I've seen @garronej using this thechnique quite extensively so might fit the repo

```
const target: string | undefined = (() => {
  if (unzipSubPath) {
    if (!entry.sfileName.startsWIth(unzipSubPath)) {
      zipfile.readEntry()
      return
    }
    return path.join(targetFolder, entry.filename.substring(unzipSubPath.length)
  }
  return path.join(targetFolder, entry.filename)

Reply:
I'd write 
```ts
const targetPath: string | undefined = (() => {
  if (unzipSubPath !== undefined) {
    if (!entry.sfileName.startsWIth(unzipSubPath)) {
      return undefined;
    }
    return path.join(targetFolder, entry.filename.substring(unzipSubPath.length)
  }
  return path.join(targetFolder, entry.filename)
})();

if( targetPath === undefined ){
          zipfile.readEntry()
}
```

I avoid side effects in closures.  I never use non boolean as boolean and I aways explicitely `return undefined` otherwise yes.

Reply:
I am not a big fan of this inplace mutation either tbf. Was a bit of mix of "fixing it ASAP" and "yes it works". 😗